hawkbit-repository/hawkbit-repository-core/src/main/java/org/eclipse/hawkbit/repository/jpa/TimestampCalculator.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2015 Bosch Software Innovations GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.repository.jpa;
+
+import java.time.Duration;
+import java.time.Instant;
+
+import org.eclipse.hawkbit.repository.TenantConfigurationManagement;
+import org.eclipse.hawkbit.repository.jpa.model.helper.TenantConfigurationManagementHolder;
+import org.eclipse.hawkbit.tenancy.configuration.DurationHelper;
+import org.eclipse.hawkbit.tenancy.configuration.TenantConfigurationKey;
+
+/**
+ * Calculates non-persistent timestamps , e.g. the point a time a
+ * target is declared as overdue.<br>
+ * Therefore tenant specific configuration may be considered.
+ *
+ */
+public class TimestampCalculator {
```

Comment:
Converting it to a class with static methods, would make it not testable in the RSQLUtilityTest, which mocks the TenantConfigurationManagement that needs to be injected in that test.